### PR TITLE
Update dependencies (v2)

### LINKED ISSUE #312: Engine.io-client xmlhttprequest-ssl high severity vulnerabilities
Saw this high severity notice generated by Dependabot for a few packages I have that are using airtap.

----

xmlhttprequest-ssl
 Open	GitHub opened this alert 8 hours ago
 Dependabot cannot update xmlhttprequest-ssl to a non-vulnerable version
The latest possible version that can be installed is 1.5.5 because of the following conflicting dependency:

airtap@4.0.3 requires xmlhttprequest-ssl@~1.5.4 via engine.io-client@3.3.2
The earliest fixed version is 1.6.2.

View logs or learn more about troubleshooting Dependabot errors.

1 xmlhttprequest-ssl vulnerability found in package-lock.json 8 hours ago
Remediation
Upgrade xmlhttprequest-ssl to version 1.6.2 or later. For example:

"dependencies": {
  "xmlhttprequest-ssl": ">=1.6.2"
}
or…
"devDependencies": {
  "xmlhttprequest-ssl": ">=1.6.2"
}
Always verify the validity and compatibility of suggestions with your codebase.

Details
CVE-2020-28502
high severity
Vulnerable versions: < 1.6.2
Patched version: 1.6.2
This affects the package xmlhttprequest before 1.7.0; all versions of package xmlhttprequest-ssl. Provided requests are sent synchronously (async=False on xhr.open), malicious user input flowing into xhr.send could result in arbitrary code being injected and run.

### PR DESCRIPTION
**Pull request by kibertoad**
fixes #312

There are two major updates, `engine.io` and `engine.io-client`. The only significant changes are on server side, client was only updated to accommodate for it. Here is the list of potentially breaking changes:

* `handlePreflightRequest` removed from CORS _(was not used in the airtap)_;
* In Engine.IO v3, the io cookie was sent by default. This cookie can be used to enable sticky-session, which is required when you have several servers _(doesn't seem to be relevant for airtap)_;
* perMessageDeflate is now disabled by default _(shouldn't make much difference for airtap)_;
* Support for Node.js 8 was dropped _(airtap is 10+)_;
* the syntax of the "wsEngine" option is updated _(airtap does not use wsEngine option)_.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,23 +49,23 @@
...
     "abstract-browser": "^1.0.0",
-    "cross-env": "^7.0.2",
-    "depcheck": "^1.2.0",
-    "hallmark": "^3.0.0",
+    "cross-env": "^7.0.3",
+    "depcheck": "^1.4.2",
+    "hallmark": "^3.1.0",
     "nyc": "^15.1.0",
     "simple-get": "^4.0.0",
     "standard": "^14.3.4",
```
Comment (kibertoad) [inline package.json:66]: There is a major update for this, but as this poses no security threat whatsoever, I let it be.

**Comment by kibertoad** [comment]
CI seems to be broken, btw. Would you be open to a PR, switching from Travis to GitHub Actions?

**Comment by vweevers** [comment]
Thanks so much for this! Limiting the amount of updates makes this easier to review and merge than #315. As for the changes:

> * In Engine.IO v3, the io cookie was sent by default. This cookie can be used to enable sticky-session, which is required when you have several servers _(doesn't seem to be relevant for airtap)_;

Correct. Airtap does run multiple servers, but it's one per browser, so a browser only ever connects to one server.

> * `handlePreflightRequest` removed from CORS _(was not used in the airtap)_;
> * perMessageDeflate is now disabled by default _(shouldn't make much difference for airtap)_;
> * Support for Node.js 8 was dropped _(airtap is 10+)_;
> * the syntax of the "wsEngine" option is updated _(airtap does not use wsEngine option)_.

👍

**Comment by vweevers** [comment]
> CI seems to be broken, btw. Would you be open to a PR, switching from Travis to GitHub Actions?

That would be most welcome. Thanks!

**Comment by vweevers** [comment]
I do have two remaining questions - same as https://github.com/airtap/airtap/pull/315#discussion_r667431712:

1. Is latest `engine.io-client` using ES6+ features? Which is why we currently pin to 3.3.2.
2. Does it support the same browsers (versions)?

**Comment by kibertoad** [comment]
@vweevers I see classes and default arguments, I take it that means "yes" on the first one?
How can I check supported versions?

**Comment by vweevers** [comment]
> I see classes and default arguments, I take it that means "yes" on the first one?

In that case we will (unfortunately) need to transpile the client bundle to ES5 (https://github.com/airtap/airtap/issues/276). E.g. by adding a babel transform to [this browserify instance](https://github.com/airtap/airtap/blob/cc1789d7ed7c41edec19d95e5c8ed6a018c1ea21/lib/content-server.js#L36).

> How can I check supported versions?

If not documented by engine-io (?) then we can run tests in Sauce Labs. Which would be good anyway. To do so, clone [`airtap/sauce`](https://github.com/airtap/sauce), run `npm i airtap/airtap#kibertoad:chore/deps-2021-09-05 --save-dev` and open a draft PR on that repo. Which won't immediately run tests for lack of access to secrets, but I will trigger them). This will cover a [whole bunch of browsers](https://github.com/airtap/sauce/blob/e072820f695f2ce2833a5026a3f83678f424f9a2/test.js#L9-L16).

**Comment by vweevers** [comment]
I see that `engine-io-client` has a [dist](https://github.com/socketio/engine.io-client/tree/master/dist) folder, perhaps it's already transpiled?

**Pull request closed without merging**